types/fsp_test.go

```diff
@@ -0,0 +1,110 @@
+package types
```

Comment:
Please add the license at the beginning of a file.

Reply:
done.